### DISCUSSION TITLE
Is it possible to add custom normal maps and roughness textures?

### DISCUSSION THREAD

**Discussion by Pio64** [Try/OpenGothic#537] [General]
I know that it's possible to use any texture pack for Gothic, but what about normal maps? As far as I know that can only be done with the DX11 mod. It would also be interesting to be able to provide more textures like roughness and metallic.

For example I've noticed that windows don't look like glass (they are not very reflective). Maybe this could be improved with a roughness texture?
![image](https://github.com/Try/OpenGothic/assets/51057487/9f521250-45d8-4ac5-9c95-cfbe02398e79)

**Comment by Try** [comment]
> Maybe this could be improved with a roughness texture?

In theory there is already 'reflectivity' functionality in original game engine (not fully implemented in opengothic, but will have to implement at some point): `material::environment_mapping` and `material::environment_mapping_strength`.
You may spot effect on metallic parts of ship-cannons in vanilla. 

Naturally that will require some modding, to decouple this window into separated material.

On top of that there is quite a challenge on engine side, to build setup that can do non-trivial materials on hi-end gpu and remain valid for low-end platforms and mid-end with 2k/4k screens - currently do RnD here, yet it's very hard so far
